proxy/proxytest/proxytest.go

```diff
@@ -21,6 +21,10 @@ type TestProxy struct {
 	server  *httptest.Server
 }
 
+func WithParamsAndRoutingOptions(fr filters.Registry, proxyParams proxy.Params, o routing.Options, routes ...*eskip.Route) *TestProxy {
```

Comment:
Maybe just make `NewTestProxy` exported

Reply:
This will change much more and maybe we can get rid of all the different `With*` funcs in proxytest altogether in this case. . I think it's better as a separate PR.

Reply:
I meant rename `newTestProxy`  to `NewTestProxy`, we should have less of `WithWhatever` not more. But this not a big deal, lets have it like you propose.